examples/ssh/opkssh/cli.go

```diff
@@ -0,0 +1,308 @@
+
+func (p *SimpleFilePolicyEnforcer) ReadPolicyFile() (string, []string, error) {
+	info, err := os.Stat(p.PolicyFilePath)
+	if err != nil {
+		return "", nil, err
+	}
+	mode := info.Mode()
+
+	// Only the owner of this file should be able to write to it
+	if mode.Perm() != fs.FileMode(0600) {
+		return "", nil, fmt.Errorf("policy file has insecure permissions, expected (0600), got (%o)", mode.Perm())
+	}
+
+	content, err := os.ReadFile(p.PolicyFilePath)
+	if err != nil {
+		return "", nil, err
+	}
+	rows := strings.Split(string(content), "\n")
+
+	for i := range rows {
+		row := strings.Fields(rows[i])
+		if len(row) > 1 {
+			email := row[0]
+			allowedPrincipals := row[1:]
+			return email, allowedPrincipals, nil
+		}
+	}
+	return "", nil, fmt.Errorf("policy file contained no policy")
+}
+
+func (p *SimpleFilePolicyEnforcer) CheckPolicy(principalDesired string, pkt *pktoken.PKToken) error {
+	allowedEmail, allowedPrincipals, err := p.ReadPolicyFile()
+	if err != nil {
+		return err
+	}
+	email, err := pkt.GetClaim("email")
+	if err != nil {
+		return err
+	}
+	if string(email) == allowedEmail {
+		if slices.Contains(allowedPrincipals, principalDesired) {
+			// Access granted
+			return nil
+		} else {
+			return fmt.Errorf("no policy to allow %s to assume %s, check policy config in %s", email, principalDesired, p.PolicyFilePath)
+		}
+	} else {
+		return fmt.Errorf("no policy for email %s, allowed email is %s, check policy config in %s", email, allowedEmail, p.PolicyFilePath)
+	}
+}
+
+func CreateSSHCert(client *parties.OpkClient, signer crypto.Signer, alg jwa.KeyAlgorithm, gqFlag bool, principals []string) ([]byte, []byte, error) {
+	pkt, err := client.OidcAuth(signer, alg, map[string]any{}, gqFlag)
+	cert, err := sshcert.BuildPktSshCert(pkt, principals)
+	if err != nil {
+		return nil, nil, err
+	}
+	caSigner, err := ssh.NewSignerFromSigner(signer)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mas, err := ssh.NewSignerWithAlgorithms(caSigner.(ssh.AlgorithmSigner), []string{ssh.KeyAlgoECDSA256})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sshcert, err := cert.SignCert(mas)
+
+	if err != nil {
+		return nil, nil, err
+	}
+	certBytes := ssh.MarshalAuthorizedKey(sshcert)
+
+	seckeySsh, err := ssh.MarshalPrivateKey(signer, "openpubkey cert")
+	if err != nil {
+		return nil, nil, err
+	}
+	seckeySshBytes := pem.EncodeToMemory(seckeySsh)
+
+	return certBytes, seckeySshBytes, nil
+}
+
+func WriteKeys(seckeyPath string, pubkeyPath string, seckeySshPem []byte, certBytes []byte) error {
+	// Write ssh secret key to filesystem
+	err := os.WriteFile(seckeyPath, seckeySshPem, 0600)
+	if err != nil {
+		return err
+	}
+	certBytes = append(certBytes, []byte(" "+"openpubkey")...)
+
+	// Write ssh public key (certificate) to filesystem
+	err = os.WriteFile(pubkeyPath, certBytes, 0777)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func FileExists(fPath string) bool {
+	_, error := os.Open(fPath)
+	return !errors.Is(error, os.ErrNotExist)
```

Comment:
This seems to be the golang standard way of checking if a file exists. It is ok that we bury the error because the only error we want to know is if the file exists.

Reply:
yes it's the way people usually do this, but they do not usually put it in its own function and bury the error. You can keep it this way but please make this function private, we should not be encouraging its use.

Reply:
Strongly disagree that this is not a clearer way of doing this. We should 100% encourage this!
I made it private because it isn't worth debating this now and FileExists should be at a much higher level so it can be used generally in the repo.